### PR TITLE
added lobby filtering by format

### LINKED ISSUE #5950: Filter game lobbies
This is more of a suggestion than an issue, but it would be very helpful to be able to filter game lobbies based on Game Type (Standard, Startup, Snapshot, etc). Apologies in advance if there's a better place than here!

Implementation could be varied.  My personal preference would be through the User Profile.  But showing it in the Play Lobby screen would make sense as well.

### PR DESCRIPTION
**Pull request by kjchiu**
resolves #5950

### CODE REVIEW

File: src/cljs/nr/appstate.cljs [hunk 1]
```diff
@@ -8,6 +8,22 @@
...
+  (let [default-visible-formats #{"standard"
+                                 "system-gateway"
+                                 "startup"
+                                 "eternal"
+                                 "snapshot"
+                                 "snapshot-plus"
+                                 "classic"
+                                 "casual"}
+       serialized (get-local-value "visible-formats" "")
+       ]
```
Comment (NoahTheDuke) [inline src/cljs/nr/appstate.cljs:23]: Hanging paren

File: src/cljs/nr/appstate.cljs [hunk 2]
```diff
@@ -8,6 +8,22 @@
   (let [rtn (js->clj (.getItem js/localStorage k))]
     (if (nil? rtn) default-value rtn)))
 
+(defn- load-visible-formats
+  "Loading visible formats from localStorage. Accounting for the fact that js->clj doesn't handle sets"
+  []
+  (let [default-visible-formats #{"standard"
+                                 "system-gateway"
```
Comment (NoahTheDuke) [inline src/cljs/nr/appstate.cljs:15]: Indentation in this block

File: src/cljs/nr/gamelobby.cljs [hunk 3]
```diff
@@ -320,13 +321,21 @@
...
   (if (= "tournament" (:room (first games)))
     games
     (let [blocked-games (filter #(blocked-from-game user %) games)
-          blocked-users (get-in user [:options :blocked-users] [])]
-      (filter #(blocking-from-game blocked-users %) blocked-games))))
+          blocked-users (get-in user [:options :blocked-users] [])
+          is-visible (some-fn
+                       #(contains? (get % :players) (:username user))
+                       (every-pred (partial blocking-from-game blocked-users) (partial hidden-formats visible-formats)))]
+      (filter #(is-visible %) blocked-games))))
```
Comment (NoahTheDuke) [inline src/cljs/nr/gamelobby.cljs:338]:
```suggestion
      (filter is-visible blocked-games))))
```

File: src/cljs/nr/gamelobby.cljs [hunk 4]
```diff
@@ -361,7 +370,28 @@
...
            [game-row (assoc game :current-game @gameid :password-game password-game :editing editing)])))]))
 
-(defn games-list-panel [s games gameid password-gameid user]
+(defn format-visible? [slug] (contains? (:visible-formats @app-state) slug))
+
+(defn format-toggle [slug]
+  (r/with-let [id (str "filter-" slug)]
+    [:div
+     [:input.visible-formats
+      {
```
Comment (NoahTheDuke) [inline src/cljs/nr/gamelobby.cljs:379]: Mind cleaning up the spacing and indentation in this block?

File: src/cljs/nr/gamelobby.cljs [hunk 5]
```diff
@@ -385,9 +415,17 @@
...
-     [room-tab user s games "tournament" (tr [:lobby.tournament "Tournament"])]
+     [:div#filter.dropdown
+      [:a.dropdown-toggle {:href "" :data-toggle "dropdown"}
+       "Filter"
+       [:b.caret]]
+       [:div.dropdown-menu.blue-shade
+        (doall (for [[k] slug->format]
+                 ^{:key k}
+                 [format-toggle k (contains? visible-formats k)]))]]
+     [room-tab user s games "casual" (tr [:lobby.casual "Casual"])]
```
Comment (NoahTheDuke) [inline src/cljs/nr/gamelobby.cljs:426]: Why change the order of the tabs?

File: src/css/lobby.styl [hunk 6]
```diff
@@ -86,18 +86,19 @@
         padding: 0 15px 15px
 
         .rooms
-            overflow-y: auto
+            overflow-y visible;
```
Comment (NoahTheDuke) [inline src/css/lobby.styl:89]: Missing a colon here?

Comment (NoahTheDuke) [reply src/css/lobby.styl:89]: and I don't think you need the semi-colon, cuz stylus will handle that for you.

File: src/cljs/nr/gamelobby.cljs (hunk 5 above)
Comment (kjchiu) [reply src/cljs/nr/gamelobby.cljs:426]: because of the change from float right to flexbox

Comment (NoahTheDuke) [reply src/cljs/nr/gamelobby.cljs:426]: Cool, thank you.

File: src/css/lobby.styl (hunk 6 above)
Comment (kjchiu) [reply src/css/lobby.styl:89]: how is stylus not showing an error in the build o.O?

**Pull request merged**